File: bot.js
var Discord = require('discord.io');
var logger = require('winston');
// Configure logger settings
logger.remove(logger.transports.Console);
logger.add(new logger.transports.Console, {
	colorize: true
});
logger.level = 'debug';
// Initialize Discord Bot
var bot = new Discord.Client({
	token: process.env.BOT_TOKEN,
	autorun: true
});
bot.on('ready', function (evt) {
	logger.info('Connected');
	logger.info('Logged in as: ' + bot.username);
});
// onGameChange hook
var usersStatus = new Map();
bot.on('presence', function (user, userID, status, game, event) {
	logger.info(user + ' changed!\n' + JSON.stringify(event));
	// bot.sendMessage({ to: '456917371882307605', message: user + ' changed somehow!'});
	if (game != null) {
		if (!usersStatus.has(userID) || usersStatus.get(userID) != game.name) {
			bot.sendMessage({ to: '456917371882307605', message: user + ' отправился в ' + game.name + '!' });
		}
		usersStatus.set(userID, game.name);
	}
	if (game == null && usersStatus.has(userID)) {
		usersStatus.delete(userID);
	}
});

File: bot.py
import discord, os

TOKEN = os.environ.get('TOKEN')

client = discord.Client()

@client.event
async def on_message(message):
    # we do not want the bot to reply to itself
    if message.author == client.user:
        return

    if message.content.startswith('!hello'):
        msg = 'Hello {0.author.mention}'.format(message)
        # await client.send_message(message.channel, msg)
        await message.channel.send(msg)

@client.event
async def on_member_update(before, after):
    if str(before.activity) == 'None' and str(after.activity) != 'None':  
        await after.guild.system_channel.send("Пэздюк " + after.display_name + " поехал в " + str(after.activity))

@client.event
async def on_ready():
    print('Logged in as')
    print(client.user.name)
    print(client.user.id)
    print('------')

client.run(TOKEN)
